File: BOJ(Java 공부용)/10816.java
import java.io.*;
import java.util.Arrays;
public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine().split(" ")[0]);
        int[] a = new int[n];
        String[] input2 = br.readLine().split(" ");
        for(int i=0;i<n;i++) {
            a[i] = Integer.parseInt(input2[i]);
        }
        Arrays.sort(a);
        int m = Integer.parseInt(br.readLine().split(" ")[0]);
        String[] input3 = br.readLine().split(" ");
        for(int i=0;i<m;i++) {
            int lower = lowerbound(0,n-1,a,Integer.parseInt(input3[i]));
            int upper = upperbound(0,n-1,a,Integer.parseInt(input3[i]));
            if(a[upper] == Integer.parseInt(input3[i])) upper++;
            upper -= lower;
            bw.write(upper + " ");
        }
        bw.flush();
        bw.close();
    }
    private static int lowerbound(int left,int right,int[] a,int num) {
        while(left<right) {
            int mid = (left+right) / 2;
            if(a[mid]<num) left = mid+1;
            else right = mid;
        }
        return right;
    }
    private static int upperbound(int left,int right,int[] a, int num) {
        while(left<right) {
            int mid = (left + right) / 2;
            if(a[mid]<=num) left = mid + 1;
            else right = mid;
        }
        return right;
    }
}

File: BOJ(Kotlin)/10816.kt
import com.sun.org.apache.xpath.internal.operations.Bool
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

fun main(args : Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val n = br.readLine().toInt()
    var st = StringTokenizer(br.readLine())
    var map = mutableMapOf<Int,Int>()
    for (i in 0 until n) {
        val num = st.nextToken().toInt()
        if (map[num] == null) {
            map[num] = 1
        }
        else {
            var value = map[num]
            map.remove(num)
            if (value != null) {
                map[num] = value + 1
            }

        }
    }

    val m = br.readLine().toInt()
    st = StringTokenizer(br.readLine())
    for (i in 0 until m) {
        val num = st.nextToken().toInt()
        if (map[num] == null) bw.write("0 ")
        else bw.write(map[num].toString()+" ")
    }
    bw.flush()
    bw.close()
}